wonnx/src/utils.rs

```diff
@@ -79,6 +82,59 @@ impl<'a> From<&'a [i32]> for InputTensor<'a> {
     }
 }
 
+impl<'a> From<&'a [i64]> for InputTensor<'a> {
+    fn from(a: &'a [i64]) -> Self {
+        InputTensor::I64(Cow::Borrowed(a))
+    }
+}
+
+#[derive(Error, Debug)]
+pub enum TensorConversionError {
+    #[error("could not convert to the requested type becaue a value could not be represented in the target type")]
+    OutOfBoundsError,
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize)]
+#[serde(untagged)]
+pub enum OutputTensor {
+    F32(Vec<f32>),
+    I32(Vec<i32>),
+    I64(Vec<i64>),
+}
+
+impl OutputTensor {
+    pub fn as_f32(self) -> Result<Vec<f32>, TensorConversionError> {
+        match self {
+            OutputTensor::F32(fs) => Ok(fs),
+            OutputTensor::I32(ints) => ints
+                .iter()
+                .map(|i| f32::from_i32(*i).ok_or(TensorConversionError::OutOfBoundsError))
+                .collect::<Result<_, _>>(),
+            OutputTensor::I64(ints) => ints
+                .into_iter()
+                .map(|i| f32::from_i64(i).ok_or(TensorConversionError::OutOfBoundsError))
+                .collect::<Result<_, _>>(),
+        }
+    }
+
+    pub fn unwrap_f32_slice(&self) -> &[f32] {
```

Comment:
it seems to me that `unwrap_f32_slice()` is equal to `as_f32().unwrap().as_slice()` which is more readable from my point of view. I feel like as this trait is going to be client-facing and therefore will have to be maintained. I will probably prefer to keep it as the standard library. What do you think?

Reply:
I know that as_f32 might be more permissive compared to unwrap_f32_slice but, this permissiveness might be a blessing if we need to handle the type of the vector with a different type for wgsl reason.

Reply:
I guess that we can even implement TryFrom/TryInto trait for OutputTensor for f32 / i32 / i64. I feel like it would be pretty neat. But open for discussion :)

Reply:
Agreed, this should be implemented through `try_into` (I just committed a change implementing TryFrom, which is apparently the idiomatic way to do this). 

Note that some extra work was needed to support slices; just having `TryFrom<OutputTensor> for Vec<f32>` doesn't make them work automatically:

````rust
common::assert_eq_vector(
  result["Y"].try_into().unwrap().as_slice(), // Fails to compile: try_into needs type annotations
  some_vec.as_slice(),
);

// The canonical way to annotate would be:
common::assert_eq_vector(
  TryInto::<Vec<f32>>::try_into(result["Y"]).unwrap().as_slice(), // works but ugly
  some_vec.as_slice(),
);

````

Because the above is quite ugly, I therefore also implement a `TryFrom<&'a OutputTensor> for &'a [f32]`. This way, `result["Y"].unwrap_f32_slice()` becomes `&(result["Y"]).try_into().unwrap()`. The downsize is that this try_into only succeeds if the output tensor is already `OutputTensor::F32` (it can't do conversion from I32/I64 like the other `TryFrom for Vec<f32>`, because as it operates on references it has nowhere to store the result). This is a bit asymmetric but still the best solution as far as I can see.